packages/istanbul-api/lib/reporter.js

```diff
@@ -91,7 +92,12 @@ Reporter.prototype = {
         const tree = this.summarizer(coverageMap);
         Object.keys(this.reports).forEach(name => {
             const report = this.reports[name];
-            tree.visit(report, context);
+            if (name === 'html-spa') {
+                const nestedTree = libReport.summarizers.nested(coverageMap);
```

Comment:
this seems reasonable to me, we always process the `nested` list, and just flatten it if you're in the flat mode?